[BUG] Truncating a partition with a duplicate partition name will cause fe to crash

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. build a cluster with 3 FE nodes.
2. create a partitioned table
```
CREATE TABLE `site_access` (
  `event_day` date NULL COMMENT "",
  `site_id` int(11) NULL DEFAULT "10" COMMENT "",
  `city_code` varchar(100) NULL COMMENT "",
  `user_name` varchar(32) NULL DEFAULT "" COMMENT "",
  `pv` bigint(20) SUM NULL DEFAULT "0" COMMENT ""
) ENGINE=OLAP
AGGREGATE KEY(`event_day`, `site_id`, `city_code`, `user_name`)
COMMENT "OLAP"
PARTITION BY RANGE(`event_day`)
(PARTITION p1 VALUES [("0000-01-01"), ("2020-01-31")),
PARTITION p2 VALUES [("2020-01-31"), ("2020-02-29")),
PARTITION p3 VALUES [("2020-02-29"), ("2020-03-31")))
DISTRIBUTED BY HASH(`site_id`) BUCKETS 10
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false",
"compression" = "LZ4"
);
```
3. truncate partition with duplicate partition name (different letter case) 
```
truncate table `site_access` partitions (p1, P1);
```

### Expected behavior (Required)


### Real behavior (Required)
```
com.starrocks.journal.JournalInconsistentException: failed to load journal type 118
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:947) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1902) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1759) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1824) [starrocks-fe.jar:?]
Caused by: java.lang.NullPointerException
        at com.starrocks.server.LocalMetastore.replayTruncateTable(LocalMetastore.java:4549) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayTruncateTable(GlobalStateMgr.java:3321) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:540) ~[starrocks-fe.jar:?]
        ... 4 more
2023-04-18 14:28:54,906 WARN (replayer|94) [GlobalStateMgr$5.runOneCycle():1762] got interrupt exception or inconsistent exception when replay journal 1363166, will exit,
com.starrocks.journal.JournalInconsistentException: failed to load journal type 118
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:947) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:1902) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1759) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.run(GlobalStateMgr.java:1824) [starrocks-fe.jar:?]
Caused by: java.lang.NullPointerException
        at com.starrocks.server.LocalMetastore.replayTruncateTable(LocalMetastore.java:4549) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayTruncateTable(GlobalStateMgr.java:3321) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:540) ~[starrocks-fe.jar:?]
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java
+++ b/fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java
@@ -3922,7 +3922,7 @@ public class LocalMetastore implements ConnectorMetadata {
         TableName dbTbl = tblRef.getName();
 
         // check, and save some info which need to be checked again later
-        Map<String, Partition> origPartitions = Maps.newHashMap();
+        Map<String, Partition> origPartitions = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER);
         OlapTable copiedTbl;
         Database db = getDb(dbTbl.getDb());
         if (db == null) {

